src/components/dialogs/parameters/dynamicsimulation/curve/dialog/equipment-filter.jsx

```diff
@@ -151,116 +203,62 @@ const EquipmentFilter = forwardRef(
-                    };
-                    rules.push(countriesRule);
-                }
-
-                // create rule IN for nominalVoltages
-                if (nominalVoltages && nominalVoltages.length > 0) {
-                    const nominalVoltagesRule = {
-                        field: FieldType.NOMINAL_VOLTAGE,
-                        operator: OperatorType.IN,
-                        values: nominalVoltages,
-                        dataType: DataType.NUMBER,
-                    };
-                    rules.push(nominalVoltagesRule);
-                }
-
-                return {
-                    combinator: CombinatorType.AND,
-                    dataType: DataType.COMBINATOR,
-                    rules,
-                };
-            };
-
-            const expertFilter = {
-                ...getTopologyKindIfNecessary(equipmentType), // for optimizing 'search bus' in filter-server
-                type: 'EXPERT',
-                equipmentType: equipmentType,
-                rules: buildExpertRules(
-                    selectedVoltageLevelIds,
-                    selectedCountries,
-                    selectedNominalVoltages
-                ),
-            };
-
-            // evaluate by filter-server
-            return evaluateJsonFilter(studyUuid, currentNode.id, expertFilter)
-                .then((equipments) => {
-                    return equipments;
-                })
-                .catch((error) => {
-                    snackError({
-                        messageTxt: error.message,
-                        headerId: 'FilterEvaluationError',
-                    });
-                });
+        }, [studyUuid, currentNode.id, snackError]);
+
+        // build fetcher which filters equipments
+        const filteringEquipmentsFetcher = useMemo(() => {
+            const expertFilter = buildExpertFilter(
+                equipmentType,
+                selectedVoltageLevelIds,
+                selectedCountries,
+                selectedNominalVoltages
+            );
+
+            // the fetcher which evaluates a filter by filter-server
+            return evaluateJsonFilter(studyUuid, currentNode.id, expertFilter);
         }, [
             studyUuid,
             currentNode.id,
             equipmentType,
-            snackError,
             selectedVoltageLevelIds,
             selectedCountries,
             selectedNominalVoltages,
         ]);
 
+        // fetching filtered equipments
         useEffect(() => {
-            if (gridReady && countriesFilterReady) {
-                equipmentsRef.current.api.showLoadingOverlay();
-                filteringEquipmentsAsync().then((equipments) => {
-                    setEquipmentRowData(equipments);
-                    equipmentsRef.current.api.hideOverlay();
-                });
+            let ignore = false;
+            if (gridReady) {
+                equipmentsRef.current?.api.showLoadingOverlay();
```

Comment:
A small comment to explain ?

Reply:
DONE! and I have also moved the call of hideOverlay() from then block to finally block..